mcx/sample.py

```diff
@@ -476,6 +477,8 @@ def get_unravel_fn():
 
     _, unravel_fn = get_unravel_fn()
 
+    init_rhat, update_rhat = online_gelman_rubin()
+    rhat_state = init_rhat(2)
```

Comment:
You probably hard coded the number of chains for testing purposes?

Reply:
Ah silly me. Yep - that's exactly what happened. Will remove it.

Reply:
I can replace this with `rhat_state = init_state.position.shape[1]` and it works but worried that if we use a different sampler we may not have a "position" property. 

Should we have a `shape` property? Should we create an abstract class for `SamplerState` and `SamplerProperties`?